core/modules/widgets/view.js

```diff
@@ -18,19 +18,95 @@ var ViewWidget = function(parseTreeNode,options) {
 	this.initialise(parseTreeNode,options);
 };
 
+var ViewHandler = function(parseTreeNode,options) {
+	this.initialise(parseTreeNode,options);
+};
+
 /*
 Inherit from the base widget class
 */
 ViewWidget.prototype = new Widget();
 
+/*
+Inherit from the base widget class
+*/
+ViewHandler.prototype = new Widget();
+
+/*
+Render this widget into the DOM
+*/
+ViewHandler.prototype.render = function(parent,nextSibling,options) {
+	this.parentDomNode = parent;
+	this.fakeWidget = this.wiki.makeTranscludeWidget(options.title,{
+		document: $tw.fakeDocument,
+		field: options.field,
+		index: options.index,
+		parseAsInline: options.mode !== "block",
+		parentWidget: this,
+		subTiddler: options.subTiddler
+	});
+	this.text = options.text || "";
+	this.viewFormat = options.format;
+	this.fakeNode = $tw.fakeDocument.createElement("div");
+	this.fakeWidget.makeChildWidgets();
+	this.fakeWidget.renderChildren(this.fakeNode,null);
+	var textNode = this.document.createTextNode(this.text);
+	parent.insertBefore(textNode,nextSibling);
+	this.domNodes.push(textNode);
+};
+
+/*
+Selectively refreshes the widget if needed. Returns true if the widget or any of its children needed re-rendering
+*/
+ViewHandler.prototype.refresh = function(changedTiddlers) {
+	var refreshed = this.fakeWidget.refresh(changedTiddlers);
+	if(refreshed) {
+		var newText;
+		switch(this.viewFormat) {
+			case "htmlwikified":
+				newText = this.fakeNode.innerHTML;
+				break;
+			case "plainwikified":
+				newText = this.fakeNode.textContent;
+				break;
+			case "htmlencodedplainwikified":
+				newText = $tw.utils.htmlEncode(this.fakeNode.textContent);
+				break;
+		}
+		if(newText !== this.text) {
+			this.domNodes[0].textContent = newText;
+			this.text = newText;
+		}
+	}
+	return refreshed;
+};
```

Comment:
Hmmm, My gut-feelings tells me, there is something strange going on. -- ViewHandler is a widget. So should be in it's own file?

Reply:
Hi @pmario , now ViewHandler isn't a widget anymore